### PR TITLE
[YUNIKORN-2568] Move all xxxEvents types to objects/events

### PR DESCRIPTION
**Pull request by pbacsko**
### What is this PR for?
Move the `xxxEvents` types to `objects\events`, because `objects` already contains too many files.

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2568

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pkg/scheduler/objects/allocation_ask.go [hunk 1]
```diff
@@ -30,6 +30,7 @@ import (
 	"github.com/apache/yunikorn-core/pkg/events"
```
Comment (chia7712) [inline pkg/scheduler/objects/allocation_ask.go:30]: maybe we can move the classes from `events` to `events/systems`? That can eliminate the reference conflicts.

**Review by chia7712 — COMMENTED** [review]
@pbacsko thanks for this patch. one small comment is left

File: pkg/scheduler/objects/allocation_ask.go (hunk 1 above)
Comment (pbacsko) [reply pkg/scheduler/objects/allocation_ask.go:30]: I don't want to introduce too much changes. I've been thinking about a better package name inside `objects`, but everything else is too long or not specific enough.

File: pkg/scheduler/objects/events/application_events.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+*/
+
+package events
+
+import (
+	"fmt"
+
+	"github.com/apache/yunikorn-core/pkg/common"
+	"github.com/apache/yunikorn-core/pkg/common/resources"
+	evt "github.com/apache/yunikorn-core/pkg/events"
```
Comment (chia7712) [inline pkg/scheduler/objects/events/application_events.go:26]: Do we need this alias? it seems there is no conflict

**Review by chia7712 — COMMENTED** [review]
@pbacsko overall +1 except for one small question

File: pkg/scheduler/objects/events/application_events.go (hunk 2 above)
Comment (pbacsko) [reply pkg/scheduler/objects/events/application_events.go]: Correct. Fixed.

**Review by chia7712 — APPROVED** [review]

**Pull request closed without merging**